### PR TITLE
Changed severity and priority functionality

### PR DESCRIPTION
**Pull request by Andrei-Treil**
Thank you for making a pull request to the UMass CS320 S22 Track1 Repo

Please do not approve or submit this request until the following have been filled out and checked: 

## Describe your changes
Changed log detail controller method to take in boolean values for all priority and severity fields. 
These fields are required for all calls to the log details api method.

## Link to relevant github issue
No official issue on github, was discussed on zoom (4/27)

## Checklist before submission
- [x] I have performed a self-review of my code
- [x] I have requested a review from a relevant classmate
- [x] I have checked the entire app builds clean locally without errors or warnings
- [ ] I have added any new tests to the appropriate directories (tests/*featurename*)
- [ ] I have waited for Jenkins status tests to finish and pass

### CODE REVIEW

File: backend/src/main/java/org/track1/clog/controller/LogDetailController.java [hunk 1]
```diff
@@ -54,21 +54,55 @@ public ResponseEntity<List<LogDetail>> getByGlobalInstanceId(@RequestParam(requi
...
             final List<Predicate> predicates = new ArrayList<Predicate>();
-            if (priority_low != null){
-                predicates.add(builder.greaterThanOrEqualTo(root.get("priority"), priority_low));
+            //predicates for each type of severity
+            Predicate predInfo = builder.between(root.get("severity"), 10, 29);
+            Predicate predWarn = builder.between(root.get("severity"),30,49);
+            Predicate predErr = builder.greaterThanOrEqualTo(root.get("severity"),50);
+            Predicate allSevPred = builder.disjunction();
+
+            //predicates for each type of severity
```
Comment (klin17) [inline backend/src/main/java/org/track1/clog/controller/LogDetailController.java:66]: Do you mean "predicates for each type of priority"?

Comment (Andrei-Treil) [reply backend/src/main/java/org/track1/clog/controller/LogDetailController.java:67]: No, allSevPred is the combination of all severity predicates, allPrioPred is for priority

**Comment by klin17** [comment]
Would it be possible to make the severity/priority params not required? You could use default values of False

Having it required later on is actually okay but right now merging this would break the front end because the front end doesn't currently use these params so all the API calls to Log Details would no longer be valid.

**Comment by Andrei-Treil** [comment]
Fixed the issue in latest commit

**Review by klin17 — APPROVED** [review]

**Review by mbk9004 — APPROVED** [review]

**Pull request merged**